src/libraries/System.Private.CoreLib/src/System/Globalization/TextInfo.WebAssembly.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Globalization
+{
+    internal static unsafe class TextInfoInterop
+    {
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern unsafe void ChangeCaseInvariantJS(out string exceptionMessage, char* src, int srcLen, char* dstBuffer, int dstBufferCapacity, bool bToUpper);
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern unsafe void ChangeCaseJS(out string exceptionMessage, in string culture, char* src, int srcLen, char* dstBuffer, int dstBufferCapacity, bool bToUpper);
+    }
+
+    public partial class TextInfo
+    {
+        internal unsafe void JsChangeCase(char* src, int srcLen, char* dstBuffer, int dstBufferCapacity, bool toUpper)
+        {
+            Debug.Assert(!GlobalizationMode.Invariant);
+            Debug.Assert(!GlobalizationMode.UseNls);
```

Comment:
GlobalizationMode.UseNls is Windows-specific property. Will it be ever true for wasm-specific code ?

Reply:
Actually not, after a second glance, it makes sense. We check here that for sure the function was not invoked with Windows-specific condition. Actually the only thing I would change in the asserts is adding `Debug.Assert(!GlobalizationMode.Hybrid)` to the other `TextInfo.*.cs` files to make sure noone will call them in hybrid mode.